napari/utils/colormaps/colormap.py

```diff
@@ -245,18 +245,20 @@ def background_as_minimum_dtype(self, dtype: np.dtype) -> int:
         int
             The background converted.
         """
-        return _cast_labels_data_to_texture_dtype(
-            np.array([self.background_value]).astype(dtype), self
-        )[0]
+        return int(
+            _cast_labels_data_to_texture_dtype(
+                np.array([self.background_value]).astype(dtype), self
+            )[0]
+        )
 
-    def map(self, values, apply_selection=True) -> np.ndarray:
+    def map(self, values, raw_colormap=False) -> np.ndarray:
```

Comment:
I don't like having a public argument here (which at a minimum should be kwarg-only btw) just to deal with caching. Instead, we could make a sub-function `_map_for_cache(nums)` that runs the right mapping when you just want to map all numbers, and `map` calls that function if necessary. That to me seems to make more sense for the public API — map should just map numbers to colors and do the right thing for all inputs — not have special modes.

Reply:
(Let me know if you want me to try to model what that would look like.)

Reply:
Ok. I have changed this. Working on simplification of DirecColormap.